packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Alert from '../Alert/Alert';
+import PropTypes from 'prop-types';
+import TableCaption from './TableCaption';
+import classNames from 'classnames';
+
+function debounce(fn, ms) {
+  let timer;
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      timer = null;
+      fn.apply(this, arguments);
+    }, ms);
+  };
+}
+
+export const Table = ({
+  className,
+  stacked,
+  striped,
+  scrollTable,
+  scrollableCaption,
+  children,
+  ...attributeOptions
+}) => {
+  const container = useRef(null);
+  const captionID = useRef('caption-' + Math.random().toString(36).substr(2, 9));
```

Comment:
What is this logic for? Is this for generating a random id?

Reply:
Yes, we need a unique number generated for the captionID.

Reply:
You should be able to use `uniqueId` if you want a unique id for the caption

Reply:
This many not work for screens with more than one table created by different components as `uniqueId` generates sequential number unique to each component, so we'll end up having duplicates in multi-table scenario.

Reply:
Oh, I always thought that `uniqueId` would just handle that case. Have you tested this? Just even though they seem sequential they should ensure no duplicates happen, otherwise whats the point of the function?

Reply:
Yes, `uniqueId` creates unique value within a component with features like prefixing the unique value. So another component using `uniqueId` will end up with the same sequential value. `uniqueId`  can be substituted by a helper function that accepts prefixes to generates unique sequential value.

Reply:
Have you tested only using `uniqueId` instead of using that function? We use `uniqueId` across the application, and if it only was unique within a single instance of a component than alot of our components should have broken.

Reply:
Thanks, retested uniqueId and it works.